### PR TITLE
Address Auth LUA Records issues

### PR DESCRIPTION
**Pull request by chbruyand**
### Short description
This PR addresses some known LUA Records issues :
 - better check input lists
 - report lua wildcards errors
 - exposes `DNSName::getRawLabels` in lua env
 - better document LUA functions and objects

This is an effort to tick some boxes in #6693 

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by chbruyand** [comment]
Added some documentation which still has TODOs.

File: pdns/lua-record.cc [hunk 1]
```diff
@@ -356,11 +368,15 @@ static bool getLatLon(const std::string& ip, string& loc)
...
 {
+  if (wips.empty()) {
+    throw std::invalid_argument("The IP list cannot be empty");
+  }
   map<double,vector<ComboAddress> > ranked;
   double wlat=0, wlon=0;
   getLatLon(bestwho.toString(), wlat, wlon);
   //        cout<<"bestwho "<<wlat<<", "<<wlon<<endl;
   vector<string> ret;
+  g_log<<Logger::Debug<< __FILE__ << ":" << __LINE__<< " " << __func__ << " wips.size()="<< wips.size() <<endl;
```
Comment (Habbie) [inline pdns/lua-record.cc]: Please wrap these in `DLOG(..)` (unless they will be removed before merge, of course!)

File: docs/dnsupdate.rst [hunk 2]
```diff
@@ -1,519 +0,0 @@
-Dynamic DNS Update (RFC2136)
```
Comment (Habbie) [inline docs/dnsupdate.rst]: I presume removing this file was a mistake :)

File: docs/lua-records/reference/comboaddress.rst [hunk 3]
```diff
@@ -0,0 +1,112 @@
+.. _ComboAddress:
```
Comment (Habbie) [inline docs/lua-records/reference/comboaddress.rst]: Wonder if we could symlink this to the dnsdist comboaddress docs?

File: pdns/lua-record.cc (hunk 1 above)
Comment (Habbie) [reply pdns/lua-record.cc]: (I see now they do get removed later)

File: docs/dnsupdate.rst (hunk 2 above)
Comment (chbruyand) [reply docs/dnsupdate.rst]: Indeed! I restored the file.

File: docs/lua-records/reference/comboaddress.rst (hunk 3 above)
Comment (chbruyand) [reply docs/lua-records/reference/comboaddress.rst]: The content is not exactly the same as I also documented the `ComboAddressSet` functionalities.

**Comment by chbruyand** [comment]
Rebased onto master.

This PR doesn't fix all issues reported in #6693 but is already a great documentation improvement. We should merge it IMHO to avoid further PR-rebasing-related efforts. Otherwise, could we state what's *really* missing here ?

Could we have an another review ?

**Pull request merged**